src/main/java/pl/cleankod/exchange/core/domain/Account.java

```diff
@@ -22,7 +24,7 @@ public static Id of(String value) {
         }
     }
 
-    public static record Number(String value) {
+    public static record Number(@JsonProperty("number") String value) {
```

Comment:
I could even be done better, because I even mixed deserializer and serializer approach. if I had to make another round if it I would stick to annotations, and provide explicit serializer and deserializer in other contract model, as you were right it is domain object